WordPress/gutenberg
# Components: Add themeable background color

Part of #44116

## What?

Extends the `Theme` component to support an arbitrary `background` color. We then generate the rest of the necessary foreground/grayscale colors based on that.

The Button component has been updated to support these new theme colors. (Except the `disabled` and `isBusy` states, which we will address later.

## Why?

It is cumbersome and error-prone for a consumer to have to generate all of the necessary theme colors. If possible, we would prefer for automatic generation to work in the majority of cases.

## How?

See `theme/color-algorithms.ts` for the logic. 

## Testing Instructions

1. `npm run storybook:dev` and go to the stories for Button.
2. Use the Theme add-on in the toolbar to switch between some preset themes.

    <img src="https://user-images.githubusercontent.com/555336/200948641-a03fa299-decf-4b6b-974c-33babb76757e.png" alt="Theme add-on" width="200">

Note that some styles are not theme-ready yet (`disabled` and `isBusy` states) and will be addressed separately.

## Screenshots or screencast 

Added a story for the `Theme` component to see the generated colors.

https://user-images.githubusercontent.com/555336/200947785-0b70a566-1576-4d76-82a2-294e62d39d20.mp4

Added a background underlay to the theme add-on in the toolbar.

https://user-images.githubusercontent.com/555336/200947828-7d23240b-7557-48dc-9e78-30fc7fed8cdb.mp4
